tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -742,22 +742,34 @@ Tensor bert_large_post_softmax_bmm(const Tensor &input_tensor_a, const Tensor &i
 
 }
 
+CoreCoord get_falcon_matmul_grid_size(Device *device){
+    CoreCoord grid_size = device->compute_with_storage_grid_size();
+    // TODO: Remove this once there are no more hangs on 8x8 (Issue #6795)
+    if (device->arch() == ARCH::WORMHOLE_B0 and grid_size.y == 8){
+        grid_size.y = 7;
+    }
+    return grid_size;
+}
+
 /**
  * Falcon matmuls using operations::primary::matmul + program_config
  */
 Tensor falcon_fused_qkv_matmul(const Tensor &input_tensor_a, const Tensor &input_tensor_b, std::optional<const Tensor> bias, const MemoryConfig& mem_config, std::optional<const DataType> output_dtype) {
-    auto program_config = bmm_op_utils::get_mcast_1d_config(input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded());
+    CoreCoord grid_size = get_falcon_matmul_grid_size(input_tensor_a.device());
+    auto program_config = bmm_op_utils::get_mcast_1d_config(input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded(), grid_size);
     return operations::primary::matmul_1d(input_tensor_a, input_tensor_b, bias, program_config, mem_config, output_dtype);
 }
 
 Tensor falcon_selfout_matmul(const Tensor &input_tensor_a, const Tensor &input_tensor_b, std::optional<const Tensor> bias, const MemoryConfig& mem_config, std::optional<const DataType> output_dtype) {
-    auto program_config = bmm_op_utils::get_mcast_1d_config(input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded());
+    CoreCoord grid_size = get_falcon_matmul_grid_size(input_tensor_a.device());
+    auto program_config = bmm_op_utils::get_mcast_1d_config(input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded(), grid_size);
     return operations::primary::matmul_1d(input_tensor_a, input_tensor_b, bias, program_config, mem_config, output_dtype);
 }
 
 Tensor falcon_dense_4h_to_h_matmul(const Tensor &input_tensor_a, const Tensor &input_tensor_b, std::optional<const Tensor> bias, const MemoryConfig& mem_config, std::optional<const DataType> output_dtype, std::optional<bool> packer_l1_acc) {
+    CoreCoord grid_size = get_falcon_matmul_grid_size(input_tensor_a.device());
     auto program_config = bmm_op_utils::get_mcast_1d_config(
-        input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded());
+        input_tensor_a, input_tensor_b, true, std::nullopt, true, mem_config.is_sharded(), grid_size);
     std::optional<const DeviceComputeKernelConfig> config = std::nullopt;
     auto compute_kernel_config = init_device_compute_kernel_config(input_tensor_a.device()->arch(), config, MathFidelity::LoFi, true, false, packer_l1_acc.value_or(false));
     return operations::primary::matmul_1d(
```

Comment:
We should just delete all of these hard-coded matmuls and use primary matmuls instead. These were here because we didn't have primary matmuls back then.

Reply:
We will need to address that in a separate pr. I will merge these as is for now since this is blocking demo release.